components/TextFields/src/ContainedInputView/private/MDCTextControlStylePathDrawingUtils.h

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+@interface MDCTextControlStylePathDrawingUtils : NSObject
+
++ (void)addTopRightCornerToPath:(UIBezierPath *)path
```

Comment:
I'm fine with moving these methods to a category if you prefer that.

You can think of these methods as slightly less esoteric wrappers around the `-addArcWithCenter:radius:startAngle:endAngle:clockwise:` you linked to. I think it's preferable for the caller to only have to say "add the bottom left corner to the path" rather than "add a curve to the path with a start angle of `-(CGFloat)((M_PI * 3) / 2)` and an end angle of `-(CGFloat)M_PI`." Do you see what I mean?

Reply:
yep i see the value of the simpler API we are adding. 

Is there a unit/snapshot comparison test we could do that confirms our results are the same as a bezierPathWithRoundedRect?

Reply:
test not required. i just got excited.

We should add a image of the end result in the description though.